src/chessmove/action.rs

```diff
@@ -43,7 +43,7 @@ impl<'a> Action<'a> for HumanAction<'a> {
 
 impl<'a> Action<'a> for AIAction<'a> {
     fn get_piece_(&self) -> &ChessPiece {
-        &self.piece_move.chess_piece
+        self.piece_move.chess_piece
```

Comment:
@neofight78 - Thank you for doing this! All these changes look great. One question though, so basically even though I was telling that function to return type:
`&&ChessPiece` 
it was being coerced into:
`&ChessPiece`?

That's a little spooky - is the rust compiler smart enough to only that that coercion when it's not a problem - but yell at me when it is? Or will it always do as much coercion as possible, and I just need to be super careful?

Reply:
Rust will automatically dereference values when needed. So an extra & usually won't break your code, but it's cleaner of course to remove it.

Reply:
Ok cool - thank you! That is good to know! I appreciate the PR